server/libs/infra/src/repositories/partner.repository.ts

```diff
@@ -0,0 +1,63 @@
+import { IPartnerRepository, PartnerDirection } from '@app/domain';
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { PartnerEntity } from '../entities';
+
+@Injectable()
+export class PartnerRepository implements IPartnerRepository {
+  readonly logger = new Logger(PartnerRepository.name);
+
+  constructor(
+    @InjectRepository(PartnerEntity)
+    private readonly repository: Repository<PartnerEntity>,
+  ) {}
+
+  getAll(userId: string, direction: PartnerDirection): Promise<PartnerEntity[]> {
+    if (direction === 'shared-by') {
+      return this.repository
+        .createQueryBuilder('partners')
+        .select('partners.sharedBy', 'sharedBy')
+        .addSelect('partners.sharedWith', 'sharedWith')
+        .addSelect('partners.createdAt', 'createdAt')
+        .addSelect('partners.updatedAt', 'updatedAt')
+        .addSelect('partners.id', 'id')
+        .where('"sharedById" = :sharedBy', { sharedBy: userId })
+        .getRawMany();
+    } else {
+      return this.repository
+        .createQueryBuilder('partners')
+        .select('partners.sharedBy', 'sharedBy')
+        .addSelect('partners.sharedWith', 'sharedWith')
+        .addSelect('partners.createdAt', 'createdAt')
+        .addSelect('partners.updatedAt', 'updatedAt')
+        .addSelect('partners.id', 'id')
+        .where('"sharedWithId" = :sharedWith', { sharedWith: userId })
+        .getRawMany();
```

Comment:
These two conditions are exactly the same except the where clause. Can you re-write it like this instead?
```typescript
return this.repository.find({ where: { [direction === 'shared-by'? 'sharedById' : 'sharedWithId']: userId })
```

I'd also recommend changing the relationship on the entity by include the id columns as strings and the relations as `UserEntity`.